readthedocs/core/middleware.py

```diff
@@ -77,3 +79,48 @@ def __call__(self, request):
             response._csp_update = update_csp_headers[url_name]
 
         return response
+
+
+class DashboardRedirectMiddleware(MiddlewareMixin):
+    """
+    Redirect users from readthedocs.org to app.readthedocs.org.
+
+    This middleware is part of the migration process to move
+    the dashboard to app.readthedocs.org.
+    """
+
+    def process_request(self, request):
+        if not settings.RTD_EXT_THEME_ENABLED:
+            return None
+
+        # Only redirect logged out users
+        if request.user.is_authenticated:
+            return None
```

Comment:
I would redirect these users as well. The deadline was on March 11th. We already communicated this.